apps/studio/components/interfaces/Storage/StoragePolicies/StoragePolicies.tsx

```diff
@@ -73,13 +73,17 @@ export const StoragePolicies = () => {
     onError: () => {},
   })
   const { mutateAsync: updateDatabasePolicy } = useDatabasePolicyUpdateMutation()
-  const { mutate: deleteDatabasePolicy } = useDatabasePolicyDeleteMutation({
-    onSuccess: async () => {
-      await refetch()
-      toast.success('Successfully deleted policy!')
-      setSelectedPolicyToDelete(undefined)
-    },
-  })
+  const { mutate: deleteDatabasePolicy, isPending: isDeletingPolicy } =
+    useDatabasePolicyDeleteMutation({
+      onSuccess: async () => {
+        await refetch()
+        toast.success('Successfully deleted policy!')
+        setSelectedPolicyToDelete(undefined)
+      },
+      onError: (error: any) => {
+        toast.error(`Failed to delete policy: ${error.message}`)
+      },
```

Comment:
btw there's default error handlers within the mutation already so technically unnecessary unless we want to have a specific error handling behaviour based on the context

no biggie though, just an fyi